src/main/scala/com/databricks/spark/xml/XmlInputFormat.scala

```diff
@@ -285,3 +283,18 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
     }
   }
 }
+
+object XmlRecordReader {
+  trait PositionTracker extends Reader {
+    val initialPosition: Long
+
+    private[this] var offset: Long = 0
+
+    def getPosition: Long = initialPosition + offset
+
+    override def read(): Int = {
```

Comment:
My concern here is that this isn't intercepting calls to `read(byte[])` etc. How about adding CountingInputStream from Common IO? it's a wrapper that's exactly meant to tell you how many bytes have been read through the input stream. I think that's what we need, added to the start position, to really know where in the file it has read.